ckanext/switzerland/helpers/decorators.py

```diff
@@ -0,0 +1,60 @@
+from datetime import datetime, timedelta
+import functools
+from ckan.common import config
+
+import logging
+
+log = logging.getLogger(__name__)
+
+
+def ratelimit(func):
+    """limits the rate of calls with the same email
+    Can be used for any logic function that has an author_email
+    in the data_dict as mandatory field and wants to limit the
+    number of calls within a given time interval.
+    - the timeinterval and limit of calls can be set in the ckan config
+    - in case of an exceeded limit the context is set accorfingly, so
+    that the api function can react on it
+    """
+    @functools.wraps(func)
+    def inner(context, data_dict):
+        author_email = data_dict.get('author_email')
+        if not author_email:
+            raise ValueError("Email is missing")
+        now = datetime.now()
+        api_calls_per_time_and_email.append((author_email, datetime.now()))
+        for (m, t) in api_calls_per_time_and_email:
+            if (t + limit_timedelta < now):
+                api_calls_per_time_and_email.remove((m, t))
+        if len([(m, t)
```

Comment:
I guess this check should be moved up before for loop in l.26, since in l.28 you modify `api_calls_per_time_and_email` and therefore len might be smaller as a real number of requests

Reply:
No this check needs to be here after the modification but I will improve my style here so that this is better visible.